lib/cc-pico.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019-2021 Fastly, Janardhan Iyengar, Kazuho Oku
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+#include "quicly/cc.h"
+#include "quicly.h"
+
+#define QUICLY_MIN_CWND 2
+#define QUICLY_PICO_BETA 0.5
+#define QUICLY_PICO_RECOVERY_PERIOD 1000 /* in milliseconds */
+
+static uint32_t calc_bytes_per_mtu_increase(uint32_t cwnd, uint32_t ssthresh, uint32_t rtt, uint32_t mtu)
+{
+    /* Increase per byte acked is the sum of 1mtu/cwnd (additive part) and ... */
+    double increase_per_byte = (double)mtu / cwnd;
+    /* ... multiplicative part being 1 during slow start, or (1/beta - 1) * RTT during congestion avoidance. */
+    if (cwnd < ssthresh) {
+        increase_per_byte += 1;
+    } else {
+        increase_per_byte += (1 / QUICLY_PICO_BETA - 1) / QUICLY_PICO_RECOVERY_PERIOD * rtt;
```

Comment:
```suggestion
        increase_per_byte += (1 / QUICLY_PICO_BETA - 1) * rtt / QUICLY_PICO_RECOVERY_PERIOD ;
```
take-it-or-leave it suggestion -- this reads just a bit more intuitively to me.

Reply:
The reason I have coded as such is due to my fear regarding how compiler optimization works (i.e., can the constants be folded together when there is a variable in the middle?). Asking Twitter wisdom: https://twitter.com/kazuho/status/1407870483614167052.

Reply:
Keeping this as-is.

Twitter wisdom told me that if we rewrite the expression as proposed, that would require an floating point division instruction at runtime, as the ANSI C prohibits compilers from merging the two constants around a variable in this case.